src/components/selectorComponents/Checkbox.tsx

```diff
@@ -26,15 +26,14 @@ const SingleCheckbox = (props: SingleCheckboxProps) => {
   const clickHandler = () => {
     const newState = !state;
     setState(newState);
-
   };
-  //two different onClicks assigned here to achieve the effect where im able to click 
+  //two different onClicks assigned here to achieve the effect where im able to click
   //the text and the checkbox itself, but not the space between them
   return (
-    <div className={styles.checkboxTextWrap} >
-      <input type="checkbox" hidden={true}/>
-      <div className={styles.textPointer} onClick={clickHandler}>{props.text}</div>
-      <div className={styles.checkboxBoxWrap} onClick={clickHandler}>
+    <div className={styles.checkboxTextWrap} onClick={clickHandler}>
+      <input type="checkbox" hidden={true} />
```

Comment:
**nitpick:** Now you don't use the full power of this input element. Let's replace this wrapping `div` with `label`, and pass this clickHandler to `onChange` in `input` element to keep up with semantics.